jackrabbit-api/src/main/java/org/apache/jackrabbit/api/binary/BinaryDownloadOptions.java

```diff
@@ -0,0 +1,317 @@
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Specifies the options to be used when obtaining a direct download URI via
+ * {@link BinaryDownload#getURI(BinaryDownloadOptions)}.  Setting these options
+ * allows the caller to instruct the service provider that these options should
+ * be applied to the response to a request made with the URI returned.
+ * <p>
+ * To specify download options, obtain a {@link BinaryDownloadOptionsBuilder}
+ * via the {@link #builder()} method, then specify the options desired and
+ * get the object via {@link BinaryDownloadOptionsBuilder#build()}.
+ * <p>
+ * If no options are needed, use {@link BinaryDownloadOptions#DEFAULT} which
+ * instructs the implementation to use the service provider default behavior.
+ */
+@ProviderType
+public final class BinaryDownloadOptions {
+    private final String mimeType;
+    private final String encoding;
+    private final String fileName;
+    private final String dispositionType;
+
+    private BinaryDownloadOptions(final String mimeType,
+                                  final String encoding,
+                                  final String fileName,
+                                  final String dispositionType) {
+        this.mimeType = mimeType;
+        this.encoding = encoding;
+        this.fileName = fileName;
+        this.dispositionType = dispositionType;
+    }
+
+    /**
+     * Provides a default instance of this class.  Using the default instance
+     * indicates that the caller is willing to accept the service provider
+     * default behavior.
+     */
+    public static final BinaryDownloadOptions DEFAULT =
+            BinaryDownloadOptions.builder().build();
+
+    /**
+     * Returns the MIME type that should be assumed for the binary that is to be
+     * downloaded.  This value should be a valid {@code jcr:mimeType}.  This
+     * value can be set by calling {@link
+     * BinaryDownloadOptionsBuilder#withMimeType(String)} when building an
+     * instance of this class.
+     *
+     * @return A String representation of the MIME type, or {@code null} if no
+     *         MIME type has been specified.
+     * @see <a href="https://docs.adobe.com/content/docs/en/spec/jcr/2.0/3_Repository_Model.html#3.7.11.10%20mix:mimeType">
+     *     JCR 2.0 Repository Model - jcr:mimeType</a>
+     */
+    @Nullable
+    public final String getMimeType() { return mimeType; }
+
+    /**
+     * Returns the encoding that should be assumed for the binary that is to be
+     * downloaded.  This value should be a valid {@code jcr:encoding}.  This
+     * value can be set by calling {@link
+     * BinaryDownloadOptionsBuilder#withEncoding(String)} when building an
+     * instance of this class.
+     *
+     * @return A String representation of the encoding, or {@code null} if no
+     *         encoding has been specified.
```

Comment:
The encoding is a name, so I think that "String representation" is confusing here.